lib/model/model_test.go

```diff
@@ -2502,6 +2469,78 @@ func TestCustomMarkerName(t *testing.T) {
 	}
 }
 
+func TestIssue4475(t *testing.T) {
+	scanned := filepath.Join("baz", "scanned")
+	unscanned := filepath.Join("baz", "unscanned")
+	defer func() {
+		defaultFs.Remove(scanned)
+		defaultFs.Remove(unscanned)
+	}()
+
+	fd, err := defaultFs.Create(scanned)
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+	fd.Close()
+
+	dbi := db.OpenMemory()
+	m := NewModel(defaultConfig, protocol.LocalDeviceID, "syncthing", "dev", dbi, nil)
+	m.AddFolder(defaultFolderConfig)
+	m.StartFolder("default")
+	m.ServeBackground()
```

Comment:
Doesn't this (or folder start) kick off an initial scan?

Reply:
yes. the scanfolder call below is used to make sure it has finished before continuing